### PR TITLE
fix change handling in sendWithReturn when subtracting from existing out

### PR DESCRIPTION
**Pull request by chappjc**
Resolves https://github.com/decred/dcrdex/issues/812

### CODE REVIEW

**Comment by chappjc** [comment]
This is going to need careful review and thorough testing.  Still, I'm considering 0.1.4 for this given the [DCR withdraw bug](https://github.com/decred/dcrdex/issues/812) is solved by this.

**Comment by chappjc** [comment]
TODO: if the withdraw amount is the entire balance, the dcr wallet backend could use the `sweepaccount` RPC for robustness.

File: client/asset/dcr/dcr.go [hunk 1]
```diff
@@ -2195,64 +2179,85 @@ func (dcr *ExchangeWallet) signTx(baseTx *wire.MsgTx) (*wire.MsgTx, error) {
...
+			if err != nil {
+				return nil, nil, "", 0, err
+			}
+			dcr.log.Debugf("Change output size = %d, addr = %s", changeOutput.SerializeSize(), changeAddress.String())
+			changeAdded = true
+			baseTx.AddTxOut(changeOutput) // unsigned txn
+			remaining -= changeValue
 		} else {
-			changeOutput.Value -= int64(minFee + changeFees)
+			fmt.Println("DUSTING IT", changeValue, remaining, minFeeWithChange)
```
Comment (chappjc) [inline client/asset/dcr/dcr.go:2248]: will remove

File: client/asset/dcr/dcr.go [hunk 2]
```diff
@@ -2195,64 +2179,85 @@ func (dcr *ExchangeWallet) signTx(baseTx *wire.MsgTx) (*wire.MsgTx, error) {
...
-		size += changeSize
-		lastFee = feeRate * uint64(size)
-		subtractee.Value = int64(reservoir - lastFee)
+	// If change added or subtracting from an existing output, iterate on fees.
+	if changeAdded || subtractFrom >= 0 {
+		subtractee := changeOutput
+		if subtractFrom >= 0 {
+			subtractee = baseTx.TxOut[subtractFrom]
+		}
+		reservoir := remaining + uint64(subtractee.Value)
```
Comment (chappjc) [inline client/asset/dcr/dcr.go:2260]: will comment on this in the code, but this reasoning is that `reservoir` is what is available for fees, `remaining` is what is presently already free and thus would be fees, and `subtractee.Value` is additional amount that can be add to fees or absorb excess fees from `remaining`.

**Review by buck54321 — APPROVED** [review]

**Pull request merged**